lib/amq/settings.rb

```diff
@@ -13,25 +13,35 @@ module Settings
       #
       # @see AMQ::Client::Settings.configure
       def self.default
-        @default ||= {
-          # server
-          :host  => "127.0.0.1",
-          :port  => AMQ::Protocol::DEFAULT_PORT,
+        # Consider to move this defaults to client libraries (bunny, amqp, auto-generated amq-protocol client)
```

Comment:
`AMQ::Settings` only holds defaults for the assumed clients and switch between Hash and String like options processing.. In my opinion, all client specific parameters should be handled (validate, fallback to default, other conditional logic, etc.) exactly inside client code, not protocol library.

I'd like to propose to retire client specific options (`frame_max`) form `AMQ::Settings` module and leave only basic ones. "Basic" means all that standard Erlang client supports according to ["URI Specification"](https://www.rabbitmq.com/uri-spec.html) and ["URI query parameters"](https://www.rabbitmq.com/uri-query-parameters.html) official RabbitMQ documentation page.

Reply:
If all clients use the same defaults, why extract anything?

Reply:
For the sake of concerns separation. Underlying protocol library is not responsible for client specific features and should adopt them as plugins/extensions. The same is reasonable for the configuration I guess.

Reply:
If the settings are identical, are they really client-specific? What's the probability of them changing *independently* in only one client? I'd say it's 0. Please stick to the changes we've discussed," concern separation" is a fantastic way of wasting a lot of time on something that you'd never get any real benefit from.

Reply:
OK :) I don't mind to stick with the current design, just announced my thoughts. I will proceed with **bunny** and **amqp**. Could we merge the specs extension in this PR anyway?